cfgmgr/intfmgr.cpp

```diff
@@ -115,6 +117,10 @@ void IntfMgr::setIntfIp(const string &alias, const string &opCmd,
     int ret = swss::exec(cmd.str(), res);
     if (ret)
     {
+        if (allow_retry && enableIpv6Flag(alias))
+        {
+            return setIntfIp(alias, opCmd, ipPrefix, false);
+        }
         SWSS_LOG_ERROR("Command '%s' failed with rc %d", cmd.str().c_str(), ret);
     }
```

Comment:
The introduction of an additional flag complicates the setIntfIp API. The code:

```c++
return setIntfIp(alias, opCmd, ipPrefix, false);
```

is not readable if you aren't aware what "false" for this API means.

I suggest to think of a way without the flag. for example:

```suggestion
    int failed = swss::exec(cmd.str(), res);
    if (failed && ipPrefix.isV6() && opCmd == "add")
    {
        SWSS_LOG_NOTICE("Failed to assign IPv6 on interface %s with return code %d, trying to enable IPv6 and retry", alias.c_str(), failed);
        if (!enableIpv6Flag(alias))
        {
            SWSS_LOG_ERROR("Failed to enable IPv6 on interface %s", alias.c_str());
            return;
        }
        failed = swss::exec(cmd.str(), res);
    }
    if (failed)
    {
        SWSS_LOG_ERROR("Command '%s' failed with rc %d", cmd.str().c_str(), failed);
    }
```

Also, note, that I've put an additional check if the IP is IPv6 since there is no point to retry IPv4 addresses. 

What do you think?

Reply:
This seems like a cleaner approach, Updated